docs/data/joy/components/select/SelectGroupedOptions.tsx

```diff
@@ -27,7 +27,6 @@ export default function SelectGroupedOptions() {
       placeholder="Choose your animal"
       slotProps={{
         listbox: {
-          component: 'div',
```

Comment:
This demo breaks if we pass this `slotProps.listbox.component`. We can revert this but I only deleted this to ask a question to understand why passing `slotProps.listbox.component` breaks the component. I want to understand the logic between `as`, `slotProps.*.component` and `internalForwardedProps.component` but it's quite difficult to understand.